tron/core/actionrun.py

```diff
@@ -165,6 +167,12 @@ def __init__(
         self.output_path = output_path or filehandler.OutputPath()
         self.output_path.append(self.id)
         self.context = command_context.build_context(self, parent_context)
+        self.retries = retries
+        self.retry_codes = retry_codes
+        if self.retries is not None and self.retry_codes is None:
+            self.retry_codes = []
+
+        self.action_command = None
```

Comment:
Is this safe? How was this getting set before?

Reply:
it is being set later on but it's a bad practice to initialize object properties outside __init__.